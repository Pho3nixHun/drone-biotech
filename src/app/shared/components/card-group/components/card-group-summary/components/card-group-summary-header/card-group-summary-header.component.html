<div
    class="grid grid-cols-subgrid col-span-full bg-gray-300 border border-gray-400 p-3 tracking-wider"
>
    <ng-content select="div" />
</div>
