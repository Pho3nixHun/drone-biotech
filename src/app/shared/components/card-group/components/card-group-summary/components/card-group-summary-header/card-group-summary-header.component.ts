import { Component } from '@angular/core';

@Component({
    selector: 'app-card-group-summary-header',
    imports: [],
    templateUrl: './card-group-summary-header.component.html',
})
export class CardGroupSummaryHeaderComponent {}
