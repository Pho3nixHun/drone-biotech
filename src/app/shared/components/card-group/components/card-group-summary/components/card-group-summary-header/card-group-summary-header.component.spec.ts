import { ComponentFixture, TestBed } from '@angular/core/testing';

import { CardGroupSummaryHeaderComponent } from './card-group-summary-header.component';
import { Component } from '@angular/core';

@Component({
    imports: [CardGroupSummaryHeaderComponent],
    template: `
        <app-card-group-summary-header>
            <div>Should be projected</div>
            <p>Should not be projected</p>
        </app-card-group-summary-header>
    `,
})
class TestHostComponent {}

describe('CardGroupSummaryHeaderComponent', () => {
    let fixture: ComponentFixture<TestHostComponent>;
    let compiled: HTMLElement;

    beforeEach(async () => {
        await TestBed.configureTestingModule({
            imports: [TestHostComponent],
        }).compileComponents();

        fixture = TestBed.createComponent(TestHostComponent);
        compiled = fixture.debugElement.nativeElement;
    });

    // Snapshot testing
    it('should project the content and render the template correctly', () => {
        // There is no need to arrange

        // Act
        fixture.detectChanges();

        // Assert
        expect(compiled).toMatchSnapshot();
    });
});
