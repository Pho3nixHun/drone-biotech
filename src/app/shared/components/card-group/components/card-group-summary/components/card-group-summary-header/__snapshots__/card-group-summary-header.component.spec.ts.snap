// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardGroupSummaryHeaderComponent should project the content and render the template correctly 1`] = `
<div
  id="root0"
  ng-version="19.2.9"
>
  <app-card-group-summary-header>
    <div
      class="grid grid-cols-subgrid col-span-full bg-gray-300 border border-gray-400 p-3 tracking-wider"
    >
      <div>
        Should be projected
      </div>
    </div>
  </app-card-group-summary-header>
</div>
`;
